lib/public_key/src/pubkey_cert.erl

```diff
@@ -144,14 +144,23 @@ validate_time(OtpCert, UserState, VerifyFun) ->
     {'Validity', NotBeforeStr, NotAfterStr} 
 	= TBSCert#'OTPTBSCertificate'.validity,
     Now = calendar:datetime_to_gregorian_seconds(calendar:universal_time()),
-    NotBefore = time_str_2_gregorian_sec(notBefore, NotBeforeStr),
-    NotAfter = time_str_2_gregorian_sec(notAfter, NotAfterStr),
 
-    case ((NotBefore =< Now) and (Now =< NotAfter)) of
-	true ->
-	    UserState;
-	false ->
-	    verify_fun(OtpCert, {bad_cert, cert_expired}, UserState, VerifyFun)
+    % Catch invalid date format errors in the `validity` field of the certificate
+    try
+        NotBefore = time_str_2_gregorian_sec(notBefore, NotBeforeStr),
+        NotAfter = time_str_2_gregorian_sec(notAfter, NotAfterStr),
+
+        case ((NotBefore =< Now) and (Now =< NotAfter)) of
+            true ->
+                UserState;
+            false ->
+                verify_fun(OtpCert, {bad_cert, cert_expired}, UserState, VerifyFun)
+        end
+
+    % "error:function_clause" is thrown by time_str_2_gregorian_sec if the date format is not valid
+    % verify_fun only throws exceptions
+    catch error:function_clause ->
+        verify_fun(OtpCert, {bad_cert, invalid_date}, UserState, VerifyFun)
```

Comment:
Also not sure if `invalid_date` is descriptive enough here?  Is it worth having something as specific as `invalid_not_before` and `invalid_not_after`, or maybe something like `invalid_cert_expiration`?

Reply:
Changed the error message into a more meaningful one.